site/pages/retirement/[beneficiary_address]/[retirement_index].tsx

```diff
@@ -0,0 +1,64 @@
+import { GetStaticProps } from "next";
+
+import { queryKlimaRetireByIndex } from "@klimadao/lib/utils";
+import { SingleRetirementPage } from "components/pages/Retirement/SingleRetirement";
+import { loadTranslation } from "lib/i18n";
+import { IS_PRODUCTION } from "lib/constants";
+
+export const getStaticProps: GetStaticProps = async (ctx) => {
+  try {
+    if (IS_PRODUCTION) {
+      throw new Error("Not on Staging");
+    }
+
+    const { params, locale } = ctx;
+
+    if (
+      !params ||
+      (!params?.beneficiary_address && !params?.retirement_index) ||
+      (!!params.retirement_index && Number(params.retirement_index) === NaN)
+    ) {
+      throw new Error("No params found");
+    }
```

Comment:
yee as commented on the prev PR lets add types arg to GetStaticProps

Reply:
GetStaticProps Params are typed